### PR TITLE
WIP: Getting Nagios back to a working state.

### PR DESCRIPTION
**Pull request by wengel**
The pull request includes some fixes to get Nagios working. But there are some issues that needs to be discussed:
- changing lib64 to lib in directory for nagios-plugins might break other distributions
- find proper way how to separate dashboard and nagios vhost config in apache2 (if nagios-server runs on dashboard node nagios is primary vhost config on that node and dashboard gets inaccessible)

### CODE REVIEW

**Comment by tboerger** [comment]
Currently nearly all core components are not really aware of other distributions as nobody maintains compat.

For the other issue we can conflict with the dashboard role/role until we find a better solution.

:+1:

File: chef/cookbooks/nagios/templates/default/nrpe.cfg.erb [hunk 1]
```diff
@@ -13,37 +13,37 @@ include_dir=/etc/nagios/nrpe.d/
...
+command[check_load]=/usr/lib/nagios/plugins/check_load -w <%= node[:nagios][:checks][:load][:warning] %> -c <%= node[:nagios][:checks][:load][:critical] %>
+command[check_all_disks]=/usr/lib/nagios/plugins/check_disk -w 8% -c 5% -A -x /dev/shm -X nfs -i /boot
+command[check_zombie_procs]=/usr/lib/nagios/plugins/check_procs -w 5 -c 10 -s Z
+command[check_total_procs]=/usr/lib/nagios/plugins/check_procs -w 500 -c 800
+command[check_swap]=/usr/lib/nagios/plugins/check_swap -w '50%' -c '25%'
+command[check_mem]=/usr/lib/nagios/plugins/check_mem.sh -w <%= node[:nagios][:checks][:memory][:warning] %> -c <%= node[:nagios][:checks][:memory][:critical] %> -p
+command[check_chef_client]=/usr/lib/nagios/plugins/check_procs -w 1:2 -c 1:2 -C chef-client
+command[check_smtp]=/usr/lib/nagios/plugins/check_smtp -H <%= node[:nagios][:checks][:smtp_host] %>
+command[check_nginx]=/usr/lib/nagios/plugins/check_procs -w 2:3 -c 1:5 -C nginx
+command[check_sphinx]=/usr/lib/nagios/plugins/check_procs -c 1:1 -C searchd
```
Comment (vuntz) [inline chef/cookbooks/nagios/templates/default/nrpe.cfg.erb]: That looks wrong; the right location to fix it is https://github.com/crowbar/barclamp-nagios/blob/master/chef/cookbooks/nagios/recipes/client.rb#L76 where you would define the lib64 to something empty for the version of SLE you were looking at.

Comment (tboerger) [inline chef/cookbooks/nagios/templates/default/nrpe.cfg.erb]: I think it should be even better to store the full path within a config variable, not only 64 :)

Comment (vuntz) [inline chef/cookbooks/nagios/templates/default/nrpe.cfg.erb]: Agreed, that's also what I would do to clean this up.

Comment (dirkmueller) [inline chef/cookbooks/nagios/templates/default/nrpe.cfg.erb]: you could simply s,/usr/lib<%= @lib64 %>/nagios/plugins/,<%= @plugin_dir %>/,

I think

Comment (tboerger) [inline chef/cookbooks/nagios/templates/default/nrpe.cfg.erb]: > you could simply s,/usr/lib<%= @lib64 %>/nagios/plugins/,<%= @plugin_dir %>/,
> 
> I think

Something like that, yes

**Comment by wengel** [comment]
I changed the plugin directory in client nrpe template to plugin_dir which is defined in client.rb so it won't brake other distros.

**Comment by rsalevsky** [comment]
Code got moved to https://github.com/crowbar/crowbar-monitoring

**Pull request closed without merging**